framework/src/executioners/Transient.C

```diff
@@ -430,7 +430,8 @@ Transient::solveStep(Real input_dt)
 
   Real current_dt = _dt;
 
-  _problem.onTimestepBegin();
+  if (_picard_it == 0)
+    _problem.onTimestepBegin();
```

Comment:
I actually tend to move `_problem.onTimestepBegin();` outside of `solveStep`. Although the fix in CrankNicolson make this unnecessary to me, we possibly should think carefully where to put this line.

Reply:
Any comment on this? Otherwise I will leave it as it is for now.

Reply:
Leave it as it is now.